snuba/consumers/utils.py

```diff
@@ -0,0 +1,54 @@
+import logging
+import time
+
+from confluent_kafka import KafkaException
+from confluent_kafka.admin import AdminClient
+
+from snuba.utils.streams.configuration_builder import get_default_kafka_configuration
+from snuba.utils.streams.topics import Topic
+
+logger = logging.getLogger("snuba.consumer.utils")
+
+
+class InvalidTopicName(Exception):
+    pass
+
+
+def get_partition_count(topic: Topic) -> int:
```

Comment:
can we add a test? Also, if this somehow fails, won't the consumer just not work?

Reply:
> can we add a test?

yeah I can do that

> Also, if this somehow fails, won't the consumer just not work?

Right now I wrap this in a try except for any exception, and if it does fail we just log that to sentry. Since Im not sure what the right balance is for the timeout values, I figured this would be a safe way to test it out first.